connectors/katalog/pkg/connector/connector.go

```diff
@@ -177,3 +177,51 @@ func (r *Handler) deleteAsset(c *gin.Context) {
 
 	c.JSON(http.StatusOK, &response)
 }
+
+// Enables deletion of assets to katalog.
+func (r *Handler) updateAsset(c *gin.Context) {
+	// Parse request
+	var request datacatalog.UpdateAssetRequest
+	if err := c.ShouldBindJSON(&request); err != nil {
+		r.reportError(c, http.StatusBadRequest, "Error during ShouldBindJSON in updateAsset"+err.Error())
+		return
+	}
+	logging.LogStructure("UpdateAssetRequest object received:", request, &r.log, zerolog.DebugLevel, false, false)
+
+	splittedID := strings.SplitN(string(request.AssetID), "/", 2)
+	if len(splittedID) != 2 {
+		errorMessage := fmt.Sprintf("UpdateAssetRequest has an invalid asset ID %s (must be in namespace/name format)", request.AssetID)
+		c.JSON(http.StatusBadRequest, gin.H{"error": errorMessage})
+		return
+	}
+	namespace, name := splittedID[0], splittedID[1]
+
+	asset := &v1alpha1.Asset{}
+	if err := r.client.Get(context.Background(), types.NamespacedName{Namespace: namespace, Name: name}, asset); err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
```

Comment:
Analyze the error, and return invalid id code if the asset does not exist (errors.IsNotFound)

Reply:
fixed. Thanks.